src/app/auth/login.ts

```diff
@@ -113,6 +136,138 @@ export const loginWithPhoneCookie = async ({
   return kratosResult
 }
 
+export const loginUpgradeWithPhone = async ({
+  phone,
+  code,
+  ip,
+  account,
+}: {
+  phone: PhoneNumber
+  code: PhoneCode
+  ip: IpAddress
+  account: Account
+}): Promise<SessionToken | ApplicationError> => {
+  {
+    const limitOk = await checkFailedLoginAttemptPerIpLimits(ip)
+    if (limitOk instanceof Error) return limitOk
+  }
+
+  {
+    const limitOk = await checkFailedLoginAttemptPerPhoneLimits(phone)
+    if (limitOk instanceof Error) return limitOk
+  }
+
+  const validCode = await isCodeValid({ phone, code })
+  if (validCode instanceof Error) return validCode
+
+  await rewardFailedLoginAttemptPerIpLimits(ip)
+  await rewardFailedLoginAttemptPerPhoneLimits(phone)
+
+  // Scenario 1) Phone account already exists (no txn with device acct) - User logs in with device jwt then he
```

Comment:
if tthis is intentional please move it to method docs

Reply:
i'll move it to a readme